OPHD/UI/FactoryProduction.cpp

```diff
@@ -215,7 +215,7 @@ void FactoryProduction::update()
 	const int labelWidth = 120;
```

Comment:
This could/should be done as a constant in Constants/Numbers.h or Constants/Ui.h -- perhaps make it a static local?

I'm a big fan of named constants being in ALL_CAPS to make it very obvious what they are -- though a namespace like constants:: also works.

I'm open to suggestion here.

Reply:
I would advise against using ALL_CAPS, as that is typically reserved for macros. Macros do not respect namespaces, so you really don't want to ever have a collision between a macro and a standard variable or constant. There is no way to protect a macro name using namespaces, they are globally replaced regardless of namespacing. This has been a problem on a few occasions previously for OPHD/NAS2D.

The typical naming conventions I'm used to for `const` variables is either normal lowerCamelCase like regular variables, or UpperCamelCase, much like classes. I think I've used UpperCamelCase a few times in this project, though I don't there's been a consistently followed convention here.

----

In general, I'm not so fond of separating constants into a separate constants file. I would prefer constants affecting a certain component be defined within that component. Makes the code more standalone, and requires less searching around to find things. Maybe it's a little easier with Visual Studio and the code navigation capabilities, though I typically don't have those features available in the editors I use.

In this case, if the value is used repeatedly across multiple UI components, it may make sense to put it in a constants file. More longer term, I kind of wonder if we might come up with another way of specifying UI constants affecting the visual layout style. Perhaps something a bit more dynamic that could support skinning. That's by no means an immediate concern though, and I'd suggest sticking to constants for the time being.

I could go either way on the placement for this variable.

----

For constants initialized with literal values, I sometimes like to mark them as `constexpr`.

In the context of literal values, there should be no real difference between `const` and `constexpr`. Either way, if you never take the address of a the variable it might actually have storage space optimized away. I'm not sure if `static const` of a local would do that.

I generally use `const` for computed constants that depend on some kind of runtime parameter, such as the return value from a non-constexpr function. I try to use `constexpr` whenever the more strict rules can apply.

Reply:
Like I said, I'm open to suggestion -- it's been my convention to use all caps but I'm not dead set on it. Agreed that it's not consistent across the project as that hasn't been something I've enforced.

I would think UpperCamelCase makes sense here as it looks/feels like a proper name which is how I view constants.

The option to put them in a namespace and in a singular place is to make it easy to find where they are. This is really mostly for constants that are used around the code, namely string literals and values that make sense to keep in one spot for tweaking (e.g., simulation parameters, UI literals, etc). They all do follow the ALL_CAPS_WITH_UNDERSCORE convention but I'm not opposed to changing this to UpperCamelCase -- should be in a different change-set me thinks.

Anyway, I digress -- I don't have a whole lot more to comment here other than things are looking good and the attention to this is much appreciated!